src/06-cocotb/bus/bus_test.py

```diff
@@ -0,0 +1,71 @@
+import cocotb
+from cocotb.clock import Clock
+import cocotb.decorators
+from cocotb.triggers import ClockCycles, RisingEdge
+from cocotb.types import LogicArray
+import random
+
+
+class HelperSerialParallel:
+    OutWidth = 8
+    counter = 0
+    res = [0] * OutWidth
+
+    def __init__(self, dut):
+        self.dut = dut
+
+    async def generate_rnd_input(self):
+        self.dut.s_valid.value = random.randint(0, 1)
+        self.dut.s_data.value = random.randint(0, 1)
+        self.dut.m_ready.value = random.randint(0, 1)
+        await RisingEdge(self.dut.clk)
+
+    async def initialize_rst(self):
+        self.dut.aresetn.value = 0
+        await ClockCycles(self.dut.clk, 2)
+        self.dut.aresetn.value = 1
+
+    async def setup(self):
+        self.dut.s_valid.value = 0
+        self.dut.m_ready.value = random.randint(0, 1)
+
+    async def my_serial_to_parallel(self):
+        if not self.dut.aresetn.value:
+            self.res = [0] * self.OutWidth
+            self.counter = 0
+        else:
+            if self.dut.s_valid.value and self.dut.s_ready.value:
+                self.res.insert(0, self.dut.s_data.value.to_unsigned())
+                self.res = self.res[0 : self.OutWidth]
```

Comment:
```suggestion
                self.res = [self.dut.s_data.value.to_unsigned()] + self.res[0 : self.OutWidth-1]
```
Решай сам, но это больше похоже на код на верилоге :)